### DISCUSSION TITLE
contract deployed but still getting undefined in truffle console

### DISCUSSION THREAD

**Discussion by pbirbarah** [ConsenSys-archive/truffle#5167] [General Q&A]
I have downloaded truffle extension in visual studio and also downloaded ganache. I was able to deploy a contract on ganache (used truffle migrate successfully) but when I use the truffle console to get info about the contract like const bla = await contract.deployed() I get undefined, can't figure out why
![ganache pic](https://user-images.githubusercontent.com/74081842/171734192-573364f3-446b-476b-ae49-aee8e2ac9fa4.PNG)

**Comment by pbirbarah** [comment]
these are the versions I am using 
![image](https://user-images.githubusercontent.com/74081842/171735275-9e144ac5-1fac-4643-8876-40bb17303cc6.png)
 and my contract uses pragma solidity >=0.7.0 <0.9.0;

**Reply by Starmand6** [reply]
I'm also getting 'undefined' from every command I do within the Truffle console. Have deployed contract using ganache just like you. My setup is: Truffle v5.5.20 (core: 5.5.20)
Ganache v7.2.0
Solidity - 0.8.15 (solc-js)
Node v16.14.2
Web3.js v1.7.4

**Reply by kevinbluer** [reply]
Hey @pbirbarah @Starmand6, i'll confirm with the team re `undefined` being returned and logged (as i'm also seeing it). However, in theory, you should still be able to interact with the contracts despite it, for example...

```
truffle(develop)> myContract = await MyDAO.deployed()
undefined
truffle(develop)> myContract.propose(...)
```
Can you give it a try and let me know (obviously switching out contract / method names as appropriate)?

In addition, @Starmand6 you mention you're seeing it with "every command I do within the Truffle console". Does that include things like running `test` or `migrate` or just when interacting with contracts?

**Reply by Starmand6** [reply]
@kevinbluer Yes, it is only when interacting with contracts.... Gotcha, yes, you are correct. I am still able to interact with the contract, set variables, etc, even after it says undefined. I'm new to Truffle and am used to `undefined` meaning something went wrong, haha. Thank you!

**Reply by kevinbluer** [reply]
Thanks for confirming @Starmand6 and awesome that you're still able to interact! That said, I'm pretty sure it shouldn't be outputting `undefined`...confirming with the core team and will loop back here shortly 👌

**Reply by cds-amal** [reply]
That `undefined` is the current behavior @Starmand6, and it shouldn't affect the value of the assigned variable. There is an outstanding [issue in our backlog](https://github.com/trufflesuite/truffle/issues/3349).
